vortex-api/src/main/java/mil/army/usace/hec/vortex/io/NetcdfDataWriter.java

```diff
@@ -24,23 +30,44 @@ public class NetcdfDataWriter extends DataWriter {
     /* Constructor */
     NetcdfDataWriter(Builder builder) {
         super(builder);
-        String overwriteOption = options.get("isOverwrite");
-        isOverwrite = overwriteOption == null || Boolean.parseBoolean(overwriteOption);
+
+        synchronized (NetcdfDataWriter.class) {
```

Comment:
This `synchronized` lock makes sure that each `NetcdfDataWriter` instance gets initialized one at a time -- to avoid race conditions of multiple instances trying to be the first one to overwrite the destination file.

Reply:
You are getting really good at concurrency I see. Add a comment in the code above the synchronized block.